city_scrapers/spiders/il_liquor_control.py

```diff
@@ -0,0 +1,86 @@
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlLiquorControlSpider(CityScrapersSpider):
+    name = "il_liquor_control"
+    agency = "Illinois Liquor Control Commission"
+    timezone = "America/Chicago"
+    allowed_domains = ["www2.illinois.gov"]
+    start_urls = [
+        "https://www2.illinois.gov/ilcc/Divisions/Pages/Legal/"
+        "Hearing-Schedule-for-Chicago-IL-and-Springfield-IL.aspx"
+    ]
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+
+        Each tentative meeting has own page. start_url is only used to get links for the pages.
+        """
+        if response.url == self.start_urls[0]:
+            for meeting_href in response.css('div.soi-link-item a::attr(href)').extract():
+                yield response.follow(meeting_href, callback=self.parse)
+        else:
+            """
+            Extract date time and location information from each meeting date.
+            Store the info in the variables.
+            The variables are passed onto '_parse_location', '_parse_start', '_parse_end'.
+            """
+            dt_str = ' '.join([
+                x.strip() for x in response.css('div.soi-event-data::text').extract()
+            ])
+            loc_str = ' '.join([
+                x.strip() for x in response.css('div.soi-eventlocation div::text').extract()
+            ])
+            meeting = Meeting(
+                title=self._parse_title(response),
+                description='',
+                classification=BOARD,
+                start=self._parse_start(dt_str),
+                end=self._parse_end(dt_str),
+                all_day=False,
+                time_notes='',
+                location=self._parse_location(loc_str),
+                links={},
+                source=response.url,
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_title(self, response):
+        """Parse or generate meeting title."""
+        return response.css('p.soi-eventType::text').extract_first().strip()
+
+    def _parse_start(self, dt_str):
+        """Parse start datetime as a naive datetime object."""
+
+        return datetime.datetime.strptime(' '.join(dt_str.split()[:6]), '%A, %B %d, %Y %I:%M %p')
+
+    def _parse_end(self, dt_str):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None"""
+        return datetime.datetime.strptime(
+            ' '.join(dt_str.split()[:4] + dt_str.split()[7:]), '%A, %B %d, %Y %I:%M %p'
+        )
+
+    def _parse_location(self, loc_str):
+        """Parse or generate location."""
+        return {
+            "address": '{} Chicago, IL'.format(' '.join(loc_str.split()[:4])),
+            "name": "",
+        }
+
+    def _parse_links(self, item):
```

Comment:
Ignore this if it's just a work in progress, but we'll also need to pull the board meeting minutes from this page https://www2.illinois.gov/ilcc/Divisions/Pages/Legal/Meeting-minutes.aspx. Also, it looks like the agendas are eventually posted on the detail page (had to find this from an older meeting), so we'll want to scrape those as well https://www2.illinois.gov/ilcc/Events/Pages/Board-Meeting-4-17-19.aspx

One thing I'm noticing is that starting on the February 21, 2019 meeting https://www2.illinois.gov/ilcc/Events/Pages/Board-Meeting-2-21-19.aspx all of the meetings listed in the minutes also have a detail page at a standard URL. One way you could handle this is to scrape the minutes page and for all dates on or after February 21, 2019 grab the minutes link, pass it to the request, and then parse the detail page the same way you already are (example here https://github.com/City-Bureau/city-scrapers/blob/master/city_scrapers/spiders/chi_buildings.py#L56-L64). Alternatively you could just parse every date listed with minutes with defaults, but I think starting in February is your best bet

Reply:
Thanks for your feedback! 
I was navigating through to see if I can use your approach, but it seems like they sometimes delete details page. For example, 
[https://www2.illinois.gov/ilcc/Events/Pages/Board-Meeting-3-21-19.aspx](https://www2.illinois.gov/ilcc/Events/Pages/Board-Meeting-3-21-19.aspx)
does not exist. 

On the other hand, for the same meeting date, an agenda link exists.
[https://www2.illinois.gov/ilcc/Events/EventDocuments/3-21-2019%20ILCC%20Agenda.pdf](https://www2.illinois.gov/ilcc/Events/EventDocuments/3-21-2019%20ILCC%20Agenda.pdf)

I can still try to implement your approach hoping they would not delete individual meeting page in the future. Let me know what you think. Thanks!